djautotask/tests/mocks.py

```diff
@@ -17,7 +17,7 @@ def create_mock_call(method_name, return_value, side_effect=None):
 
 def init_api_rest_connection(return_value=None):
     method_name = 'djautotask.api.get_api_connection_url'
```

Comment:
I found a problem in this unit test you added for the REST switch while making my changes to the issue. You are testing a method too high in the call stack now. By testing here instead of zone info, you are just setting the cache rather than testing the logic of whether it should fetch from the cache or refresh the data from the API. I'll update this too.